examples/DGmethods/ex_001_dycoms.jl

```diff
@@ -58,100 +57,363 @@ eprint = {https://doi.org/10.1175/MWR2930.1}
+
+  rho_tot   = MPI.Reduce(rho_localtot, +, 0, MPI.COMM_WORLD)
+  U_tot     = MPI.Reduce(U_localtot, +, 0, MPI.COMM_WORLD)
+  V_tot     = MPI.Reduce(V_localtot, +, 0, MPI.COMM_WORLD)
+  W_tot     = MPI.Reduce(W_localtot, +, 0, MPI.COMM_WORLD)
+  e_tot     = MPI.Reduce(e_localtot, +, 0, MPI.COMM_WORLD)
+  qt_tot    = MPI.Reduce(qt_localtot, +, 0, MPI.COMM_WORLD)
+  qliq_tot  = MPI.Reduce(qliq_localtot, +, 0, MPI.COMM_WORLD)
+  qice_tot  = MPI.Reduce(qice_localtot, +, 0, MPI.COMM_WORLD)
+  qvap_tot  = MPI.Reduce(qvap_localtot, +, 0, MPI.COMM_WORLD)
+  T_tot     = MPI.Reduce(T_localtot, +, 0, MPI.COMM_WORLD)
+  theta_tot = MPI.Reduce(theta_localtot, +, 0, MPI.COMM_WORLD)
+
+  if mpirank == 0
+    rho_avg   = rho_tot / (size(localQ, 1) * size(localQ, 3) * nranks)
+    U_avg     = (U_tot / (size(localQ, 1) * size(localQ, 3) * nranks))/rho_avg
+    V_avg     = (V_tot / (size(localQ, 1) * size(localQ, 3) * nranks))/rho_avg
+    W_avg     = (W_tot / (size(localQ, 1) * size(localQ, 3) * nranks))/rho_avg
+    e_avg     = (e_tot / (size(localQ, 1) * size(localQ, 3) * nranks))/rho_avg
+    qt_avg    = (qt_tot / (size(localQ, 1) * size(localQ, 3) * nranks))/rho_avg
+    qliq_avg  = (qliq_tot / (size(localQ, 1) * size(localQ, 3) * nranks))
+    qice_avg  = (qice_tot / (size(localQ, 1) * size(localQ, 3) * nranks))
+    qvap_avg  = (qvap_tot / (size(localQ, 1) * size(localQ, 3) * nranks))
+    T_avg     = (T_tot / (size(localQ, 1) * size(localQ, 3) * nranks))
+    theta_avg = (theta_tot / (size(localQ, 1) * size(localQ, 3) * nranks))
+
+    @info "ρ average = $(rho_avg)"
+    @info "U average = $(U_avg)"
+    @info "V average = $(V_avg)"
+    @info "W average = $(W_avg)"
+    @info "e average = $(e_avg)"
+    @info "qt average = $(qt_avg)"
+    @info "qliq average = $(qliq_avg)"
+    @info "qice average = $(qice_avg)"
+    @info "qvap average = $(qvap_avg)"
+    @info "T average = $(T_avg)"
+    @info "theta average = $(theta_avg)"
+  end
+
+  AVG = SVector(rho_avg, U_avg, V_avg, W_avg, e_avg, qt_avg)
+  AVG_T = SVector(qliq_avg, qice_avg, qvap_avg, T_avg, theta_avg)
+
+  #fluctuations
+  for s in 1:6  
+    for e in 1:nrealelem
+      for i in 1:Nq * Nqk * Nq
+        if s == 1
+          fluctQ[i,s,e] = localQ[i,s,e] - AVG[s]
+          varQ[i,s,e]   = fluctQ[i,s,e]^2
+          fluctT[i,s,e] = thermoQ[i,s,e] - AVG_T[s]
+        elseif s < 6
+          fluctQ[i,s,e] = localQ[i,s,e] / localQ[i,1,e] - AVG[s]
+          varQ[i,s,e]   = fluctQ[i,s,e]^2
+          fluctT[i,s,e] = thermoQ[i,s,e] - AVG_T[s]
+        else
+          fluctQ[i,s,e] = localQ[i,s,e] / localQ[i,1,e] - AVG[s]
+          varQ[i,s,e]   = fluctQ[i,s,e]^2
+        end
+      end
+    end
+  end
+
+  #standard_deviation
+  rho_local_flucttot = sum(varQ[:,1,:])
+  U_local_flucttot   = sum(varQ[:,2,:])
+  V_local_flucttot   = sum(varQ[:,3,:])
+  W_local_flucttot   = sum(varQ[:,4,:])
+  e_local_flucttot   = sum(varQ[:,5,:])
+  qt_local_flucttot  = sum(varQ[:,6,:])
```

Comment:
One of the things we talked about yesterday with @kpamnany, is that these sums should probably be integral approximations (involving quadrature weights and jacobian determinants). Otherwise, you are oversampling the boundary, small cells, etc.

(same goes to the other sums in this PR)

Reply:
@jkozdon This is a good point, I neglected to think of this when I was setting up the sums. 
to fix this could we do "Jc = vgeo[ijk, _JcV, e]" and multiply each contribution by it's corresponding Jc

Reply:
This seems to be the approach taken in the DGmodel kernels. with https://github.com/climate-machine/CLIMA/blob/6b516b6cf0ef5632923a442a742e23529b161a46/src/DGmethods/DGmodel_kernels.jl#L22

Reply:
You would want to use `_M` since this is the volume mass matrix (which includes quadrature weights and Jacobian determinants). 

`JcV` is Jacobian determinant for the 1D vertical line integrals.

Reply:
Ah alright, I'm still unfamiliar with where exactly everything is so I've been using the stack integral as a reference. Would _M work for computing the averages on a plane as well ?

Reply:
No. You would need to compute it from the other metric terms. I cannot tell you what it is off the top of my head for the horizontal direction, but you can see how we get the vertical one at in the `Grids.jl` [1]

Also, planes will only be planes in physical space when the mesh is not warped in the vertical direction.

[1]: https://github.com/climate-machine/CLIMA/blob/69d298fb62c276d9dec884e54ada9ce35143f6a6/src/Mesh/Grids.jl#L293-L309

Reply:
Ah alright I'll have a look at Grids.jl then.
We'd still need to take into account the LGL quadrature weights then right ?

Reply:
Correct. The LGL weights are built into `M`, but not `JcV` (when used in for vertical indefinite integrals the weights are accounted for in the indefinite integral operator).

If you were going to define a horizontal, definite integral function I would probably build an `MH` array that included both the horizontal Jacobian determinants and the quadrature weights (like is done for `M`).

Reply:
Alright, yeah I'll see what's done with M and and try to define an MH array. I'm assuming this should go in it's own separate PR.